### ISSUE TITLE
[feature request] save and load viewer

### ISSUE THREAD

**Issue by jshenguru** [pattern-x/gemini-viewer-examples#65]
For a large dxf file, it takes about 7 seconds to load dxf, parse dxf, construct viewer, then render. In real-life use, this dxf file does not change and it is a waste of time to parse and construct viewer every time the user open up the web page.

Is it possible to provide exportViewer() so that I can save the constructed viewer along with all three.js entities in it to a file or Local Storage or IndexedDB?

In companion with the exportViewer() would be the loadViewer() which can use the exported file to construct a viewer and render it. This would improve user experience a lot.

**Comment by yanzexuan1** [comment]
It must be the second time you load this model as I can see "Loaded DxfData from indexedDb in 0.436s". For the first time, we parse the dxf file and save it to indexeddb, so the second time just load from indexeddb.

And loadEntities() for the second load, means generating threejs objects from dxf data. 

We should be able to serialize theejs scene, and directly load it. The problem is, threejs objects are large, indexeddb is not large enough to store many of them.

**Comment by jshenguru** [comment]
Yes, my application has to switch across several CAD drawings back and forth. The ideal user experience would be that when user switch to a CAD drawing the 2nd time or forward, the application should skip the loadEntities() and "adding model to scene", instead it should simply load the scene.

Regarding the threejs scene object being large, there are several options:

Option 1. Zip up the threejs scene JSON before storing into indexedDB.
Option 2. Allow user download the threejs JSON to local drive. I have already tried manually run viewer.scene.toJSON() then download. My problem is that the xdfViewer does not allow me to load that scene JSON.

**Comment by yanzexuan1** [comment]
How large is your drawings usually? 

Is it ok to create several DxfViewers, and don't destroy them if user is likely to switch back?

By seralizing the scene, we are ok to see objects, while there are other data in memory, like layouts, layers and many others. Many features won't work without these data.
